URDFRigidBodyManipulator.h missing?

Hi,

I noticed there were some nice C++ additions to the code base and tried to compile and run the test-cases (e.g. built from `systems/plants/test/testIKtraj.cpp`). But when I try to compile them, I get a bunch of "file missing" errors:
```drake-distro/drake/systems/plants/test/testIK.cpp:4:38: fatal error: URDFRigidBodyManipulator.h: No such file or directory?```

I checked and didn't find the file in the repos either. Is this some oversight or is there another reason that the file was removed?

Best,
Achim

URDFRigidBodyManipulator was removed because the functionality it provided
has been incorporated into the main RigidBodyManipulator C++ class. You've
found some tests we forgot to update. Thanks!

On Wed, Mar 18, 2015, 8:08 PM Achim Stein <notifications@github.com> wrote:

> Hi,
>
> I noticed there were some nice C++ additions to the code base and tried to
> compile and run the test-cases (e.g. built from
> systems/plants/test/testIKtraj.cpp). But when I try to compile them, I
> get a bunch of "file missing" errors:
> drake-distro/drake/systems/plants/test/testIK.cpp:4:38: fatal error:
> URDFRigidBodyManipulator.h: No such file or directory?
>
> I checked and didn't find the file in the repos either. Is this some
> oversight or is there another reason that the file was removed?
>
> Best,
> Achim
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RobotLocomotion/drake/issues/872>.
>

--- a/systems/plants/test/testIK.cpp
+++ b/systems/plants/test/testIK.cpp
@@ -1,7 +1,6 @@
 #include "RigidBodyIK.h"
 #include "RigidBodyManipulator.h"
 #include "../constraint/RigidBodyConstraint.h"
-#include "URDFRigidBodyManipulator.h"
 #include "../IKoptions.h"
 #include <iostream>
 #include <cstdlib>
@@ -11,7 +10,8 @@ using namespace std;
 using namespace Eigen;
 int main()
 {
-  URDFRigidBodyManipulator* model = loadURDFfromFile("./examples/Atlas/urdf/atlas_minimal_contact.urdf");
+  RigidBodyManipulator rbm("../examples/Atlas/urdf/atlas_minimal_contact.urdf");
+  RigidBodyManipulator* model = &rbm;
   if(!model)
   {
     cerr<<"ERROR: Failed to load model"<<endl;
@@ -39,7 +39,7 @@ int main()
   {
     return 1;
   }
-  model->doKinematics(q_sol.data());
+  model->doKinematics(q_sol);
   Vector3d com;
   model->getCOM(com);
   printf("%5.2f\n%5.2f\n%5.2f\n",com(0),com(1),com(2));

--- a/systems/plants/test/testIK.cpp
+++ b/systems/plants/test/testIK.cpp
@@ -10,7 +10,7 @@ using namespace std;
 using namespace Eigen;
 int main()
 {
-  RigidBodyManipulator rbm("../examples/Atlas/urdf/atlas_minimal_contact.urdf");
+  RigidBodyManipulator rbm("examples/Atlas/urdf/atlas_minimal_contact.urdf");
   RigidBodyManipulator* model = &rbm;
   if(!model)
   {

--- a/systems/plants/test/testIKpointwise.cpp
+++ b/systems/plants/test/testIKpointwise.cpp
@@ -1,7 +1,6 @@
 #include "RigidBodyIK.h"
 #include "RigidBodyManipulator.h"
 #include "../constraint/RigidBodyConstraint.h"
-#include "URDFRigidBodyManipulator.h"
 #include "../IKoptions.h"
 #include <iostream>
 #include <cstdlib>
@@ -10,7 +9,8 @@ using namespace std;
 using namespace Eigen;
 int main()
 {
-  URDFRigidBodyManipulator* model = loadURDFfromFile("examples/Atlas/urdf/atlas_minimal_contact.urdf");
+  RigidBodyManipulator rbm("../examples/Atlas/urdf/atlas_minimal_contact.urdf");
+  RigidBodyManipulator* model = &rbm;
   if(!model)
   {
     cerr<<"ERROR: Failed to load model"<<endl;

--- a/systems/plants/test/testIKpointwise.cpp
+++ b/systems/plants/test/testIKpointwise.cpp
@@ -9,7 +9,7 @@ using namespace std;
 using namespace Eigen;
 int main()
 {
-  RigidBodyManipulator rbm("../examples/Atlas/urdf/atlas_minimal_contact.urdf");
+  RigidBodyManipulator rbm("examples/Atlas/urdf/atlas_minimal_contact.urdf");
   RigidBodyManipulator* model = &rbm;
   if(!model)
   {

--- a/systems/plants/test/testIKtraj.cpp
+++ b/systems/plants/test/testIKtraj.cpp
@@ -1,7 +1,7 @@
 #include "RigidBodyIK.h"
 #include "RigidBodyManipulator.h"
 #include "../constraint/RigidBodyConstraint.h"
-#include "URDFRigidBodyManipulator.h"
+
 #include "../IKoptions.h"
 #include <iostream>
 #include <cstdlib>
@@ -10,7 +10,9 @@ using namespace std;
 using namespace Eigen;
 int main()
 {
-  URDFRigidBodyManipulator* model = loadURDFfromFile("examples/Atlas/urdf/atlas_minimal_contact.urdf");
+  RigidBodyManipulator rbm("../examples/Atlas/urdf/atlas_minimal_contact.urdf");
+  RigidBodyManipulator* model = &rbm;
+
   if(!model)
   {
     cerr<<"ERROR: Failed to load model"<<endl;
@@ -43,7 +45,7 @@ int main()
   int nq = model->num_positions;
   VectorXd qstar = VectorXd::Zero(nq);
   qstar(3) = 0.8;
-  model->doKinematics(qstar.data());
+  model->doKinematics(qstar);
   Vector3d com0;
   model->getCOM(com0);
   Vector4d l_hand_pt;

--- a/systems/plants/test/testIKtraj.cpp
+++ b/systems/plants/test/testIKtraj.cpp
@@ -10,7 +10,7 @@ using namespace std;
 using namespace Eigen;
 int main()
 {
-  RigidBodyManipulator rbm("../examples/Atlas/urdf/atlas_minimal_contact.urdf");
+  RigidBodyManipulator rbm("examples/Atlas/urdf/atlas_minimal_contact.urdf");
   RigidBodyManipulator* model = &rbm;
 
   if(!model)

